skeletor.el

```diff
@@ -511,9 +516,10 @@ substitution."
   (cl-assert (f-exists? src))
   (cl-assert (stringp dest))
   (make-directory dest t)
-  (->> (skeletor--dir->SkeletorTemplate src)
-       (skeletor--expand-template-paths substitutions dest)
-       (skeletor--instantiate-spec substitutions)))
+  (let ((skeletor substitutions))
```

Comment:
```suggestion
  (let ((skeletor--dynamic-substitutions substitutions))
```

Reply:
Forgot to hit submit on some other usages, but you get my drift. :)

Reply:
I feel the `--` is unecessary because users are expected to access and use it (from templates) or maybe I'm misunderstanding the purpose of `--` :/.

Also, wow, you're right. We can pass a lexical alist of allowed variables to `eval` and it works :joy:.

```lisp
(let ((foo "bar"))
  (eval '(format "%s" foo) `((foo . ,foo))))
```

If it's all right with you I'd prefer to just make the substitutions available as `subs` (or `.subs` if you'd prefer).

Reply:
@chrisbarrett 

I've removed the `skeletor` variable and now exposed the current substitutions in a lexically scoped `subs` variable. Please test and let me know if you find any issues.